cmd/ui.go

```diff
@@ -185,35 +252,33 @@ func showProgress(ctx context.Context, conf Config, execScheduler *local.Executi
 	//description in the TODO message in cmd/root.go)
 	if stdoutTTY && !noColor {
 		updateFreq = 100 * time.Millisecond
-		outMutex.Lock()
-		stdout.PersistentText = progressBarsPrint
-		stderr.PersistentText = progressBarsPrint
-		outMutex.Unlock()
-		defer func() {
-			outMutex.Lock()
-			stdout.PersistentText = nil
-			stderr.PersistentText = nil
-			if ctx.Err() != nil {
-				// Render a last plain-text progressbar in an error
-				progressBarsLastRender = []byte(renderMultipleBars(stdoutTTY, false, leftMax, pbs))
-				progressBarsPrint()
-			}
-			outMutex.Unlock()
-		}()
 	}
 
 	ctxDone := ctx.Done()
 	ticker := time.NewTicker(updateFreq)
+	sigwinch := NotifyWindowResize()
+	fd := int(os.Stdout.Fd())
 	for {
 		select {
-		case <-ticker.C:
-			barText := renderMultipleBars(stdoutTTY, true, leftMax, pbs)
-			outMutex.Lock()
-			progressBarsLastRender = []byte(barText)
-			progressBarsPrint()
-			outMutex.Unlock()
 		case <-ctxDone:
+			// FIXME: Remove this...
+			// Add a small delay to allow executors time to process
+			// the done context, so that the correct status symbol is
+			// outputted for each progress bar.
+			time.Sleep(50 * time.Millisecond)
+			renderProgressBars(false)
+			printProgressBars()
 			return
+		case <-ticker.C:
+			// Optional "polling" method, platform dependent.
+			termWidth, _, _ = GetTermSize(fd, termWidth)
```

Comment:
Shouldn't this not happen on *nix systems ?

Reply:
It doesn't :) Notice that `GetTermSize` is a no-op in `ui_unix.go`. I took this approach to avoid doing `runtime.GOOS` checks, but maybe that would be better? Let me know.

Reply:
You still have a ticker ticking and needing to the loop here. It is not that big of a problem, but if we can avoid it, I think it will be better

Reply:
Let me know if 1d28f2b4 is an improvement.

The ticker is still needed in both cases to pace the rendering, but the refactor seems cleaner. It also surprisingly fixed the need for the [delay hack](https://github.com/loadimpact/k6/pull/1332#discussion_r380129011), so I think it's a net positive. :)